docs/cow-protocol/concepts/flashloans/how-it-works.md

```diff
@@ -0,0 +1,51 @@
+---
+sidebar_position: 3
+---
+
+# How does it work?
+
+The flashloan's flow can be summarized to:
+
+1. Advance funds to the user.
+2. Invoke a function that allows the user to execute an operation with the borrowed funds.
+3. Return the funds to the lender.
+4. Verify that the full amount has been repaid.
+
+If any step fails, the entire transaction is reverted, ensuring that no funds are moved. This makes flash loans risk-free, even without collateral.
+
+All steps must occur within the same caller context. The reason it holds on to the caller context is that the user's token must be advanced into the settlement contract (so that a solver has funds to work with) and only send the swapped tokens out at the end of the settlement. By holding on to the caller's context, it is possible to revert if the tokens cannot be transferred at the end of the settlement. However, CoW Protocol cannot hold on to this context directly. To ensure that all steps execute within the same caller context, the GPv2 Settlement contract's `settle()` function is called from within the Flashloan Settlement Wrapper contract callback. Rather than directly calling the GPv2 Settlement contract, the solver first interacts with the Flashloan Settlement Wrapper contract.
```

Comment:
> By holding on to the caller's context, it is possible to revert if the tokens cannot be transferred at the end of the settlement.

This wording is a bit weird. It sounds like we decided to implement a way where we hold on to the caller context when it's actually dictated by how flashloans work.